include/language-factory.php

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * PLL_Language factory.
+ *
+ * @since 3.4
+ */
+class PLL_Language_Factory {
+	/**
+	 * Returns a language object matching the given data, looking up in cached transient.
+	 *
+	 * @since 3.4
+	 *
+	 * @param array $language_data {
+	 *     Language object properties stored as an array.
+	 *
+	 *     @type array[] $term_props      An array of language term properties. Array keys are language taxonomy names
```

Comment:
Maybe I didn't understand, but I don't think so since it's an array such as:
```php
array(
     'language'       => array(
         'term_id'          => 7,
         'term_taxonomy_id' => 8,
         'count'            => 11,
     ),
    'term_language' => array(
        'term_id'          => 11,
        'term_taxonomy_id' => 12,
        'count'            => 6,
    ),
);
```